### DISCUSSION TITLE
pimcore_iterate_block error after updating non-pimcore packages

### DISCUSSION THREAD

**Discussion by alasdair-shields** [pimcore/pimcore#17173] [Admin-Ui-Classic-Bundle]
This refers to Pimcore 10.6.9 - I don't know if other versions are affected however I'm mentioning this in case others are affected.

Yesterday (2024-06-12) I ran composer update.  Pimcore itself was NOT updated in any way - it was already at 10.6.9.  The upadate affected only supporting packages.

After the update, if a document containing **pimcore_iterate_block** was saved, on save the data got corrupted.  Any attempt to re-open the document would result in a "duplicate editable" warning.  We confirmed this by testing with the sample code from the documentation.

In the documents_editables table the 'data' value for the json 'gridblock' entry had an extra entry.  For example, 

a:3:{i:0;s:1:"1";i:1;s:1:"2";i:2;s:1:"3";}  #correct

a:4:{i:0;s:1:"1";i:1;s:1:"2";i:2;s:1:"3";i:1;s:1:"2";} #error - note that **i:1;s:1:"2";** appears twice.

I haven't been able to trace what's causing the problem. Reversing the composer update fixed the issue however I had to manually update the corrupted entries in the database before affected documents could be re-opened.

**Comment by fashxp** [comment]
did you update twig? what version do you use?

**Comment by alasdair-shields** [comment]
The twig-related updates were:

  - Upgrading twig/twig (v3.7.1 => v3.10.3)
  - Upgrading twig/string-extra (v3.7.1 => v3.10.0)
  - Upgrading twig/extra-bundle (v3.7.1 => v3.10.0)
  - Upgrading symfony/twig-bridge (v5.4.26 => v5.4.40)
  - Upgrading symfony/twig-bundle (v5.4.27 => v5.4.40)

**Reply by jdreesen** [reply]
Pimcore is currently not compatible with Twig >= 3.9.0. See: #16958
